config/v1/types_securityprofile.go

```diff
@@ -0,0 +1,427 @@
+package v1
+
+// TLSSecurityProfile defines the schema for a TLS security profile. This object
+// is used by operators to apply TLS security settings to operands.
+// +union
+type TLSSecurityProfile struct {
+	// type is one of Old, Intermediate, Modern or Custom. Custom provides
+	// the ability to specify individual TLS security profile parameters.
+	// Old, Intermediate and Modern are TLS security profiles based on:
+	//
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Recommended_configurations
+	//
+	// The profiles are intent based, so they may change over time as new ciphers are developed and existing ciphers
+	// are found to be insecure.  Depending on precisely which ciphers are available to a process, the list may be
+	// reduced.
+	//
+	// +unionDiscriminator
+	// +optional
+	Type TLSProfileType `json:"type"`
+	// old is a TLS security profile based on:
+	//
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Old_backward_compatibility
+	//
+	// and looks like this (yaml):
+	//
+	//   ciphers:
+	//     - ECDHE-ECDSA-CHACHA20-POLY1305
+	//     - ECDHE-RSA-CHACHA20-POLY1305
+	//     - ECDHE-RSA-AES128-GCM-SHA256
+	//     - ECDHE-ECDSA-AES128-GCM-SHA256
+	//     - ECDHE-RSA-AES256-GCM-SHA384
+	//     - ECDHE-ECDSA-AES256-GCM-SHA384
+	//     - DHE-RSA-AES128-GCM-SHA256
+	//     - DHE-DSS-AES128-GCM-SHA256
+	//     - kEDH+AESGCM
+	//     - ECDHE-RSA-AES128-SHA256
+	//     - ECDHE-ECDSA-AES128-SHA256
+	//     - ECDHE-RSA-AES128-SHA
+	//     - ECDHE-ECDSA-AES128-SHA
+	//     - ECDHE-RSA-AES256-SHA384
+	//     - ECDHE-ECDSA-AES256-SHA384
+	//     - ECDHE-RSA-AES256-SHA
+	//     - ECDHE-ECDSA-AES256-SHA
+	//     - DHE-RSA-AES128-SHA256
+	//     - DHE-RSA-AES128-SHA
+	//     - DHE-DSS-AES128-SHA256
+	//     - DHE-RSA-AES256-SHA256
+	//     - DHE-DSS-AES256-SHA
+	//     - DHE-RSA-AES256-SHA
+	//     - ECDHE-RSA-DES-CBC3-SHA
+	//     - ECDHE-ECDSA-DES-CBC3-SHA
+	//     - EDH-RSA-DES-CBC3-SHA
+	//     - AES128-GCM-SHA256
+	//     - AES256-GCM-SHA384
+	//     - AES128-SHA256
+	//     - AES256-SHA256
+	//     - AES128-SHA
+	//     - AES256-SHA
+	//     - AES
+	//     - DES-CBC3-SHA
+	//     - HIGH
+	//     - SEED
+	//     - "!aNULL"
+	//     - "!eNULL"
+	//     - "!EXPORT"
+	//     - "!RC4"
+	//     - "!MD5"
+	//     - "!PSK"
+	//     - "!RSAPSK"
+	//     - "!aDH"
+	//     - "!aECDH"
+	//     - "!EDH-DSS-DES-CBC3-SHA"
+	//     - "!KRB5-DES-CBC3-SHA"
+	//     - "!SRP"
+	//   tlsVersion:
+	//     minimumVersion: TLSv1.0
```

Comment:
@ironcladlou @knobunc are you sure this is right?

Reply:
The api was originally designed for ingress operator to manage tls settings of the haproxy router. The min and max versions came from:

https://cbonte.github.io/haproxy-dconv/1.8/configuration.html#5.1-ssl-min-ver
https://cbonte.github.io/haproxy-dconv/1.8/configuration.html#5.1-ssl-max-ver

Reply:
Even with the recent update, https://wiki.mozilla.org/Security/Server_Side_TLS#Old_backward_compatibility still states TLS 1.0 is the minimum version.

Reply:
I think we go with 1.0 for old to match the mozilla settings in case someone needs to support old devices.  But I do not expect many people to use it.